keywords/CouchbaseServer.py

```diff
@@ -250,3 +267,212 @@ def get_server_docs_with_prefix(self, url, bucket, prefix):
                 found_ids.append(row["$1"]["id"])
 
         return found_ids
+
+    def _wait_for_rebalance_complete(self, url):
+        """
+        Polls couchbase server tasks endpoint for any running rebalances.
+        Exits when no rebalances are in running state
+
+        /pools/default/tasks format:
+        [
+            {
+                "type": "rebalance",
+                "status": "running",
+                ...
+            }
+        ]
+        """
+
+        start = time.time()
+        while True:
+
+            if time.time() - start > keywords.constants.REBALANCE_TIMEOUT:
+                raise Exception("wait_for_rebalance_complete: TIMEOUT")
+
+            resp = self._session.get("{}/pools/default/tasks".format(url))
+            log_r(resp)
+            resp.raise_for_status()
+
+            resp_obj = resp.json()
+
+            done_rebalacing = True
+            for task in resp_obj:
+                # loop through each task and see if any rebalance tasks are running
+                task_type = task["type"]
+                task_status = task["status"]
+                log_info("{} is {}".format(task_type, task_status))
+                if task_type == "rebalance" and task_status == "running":
```

Comment:
Do you think it is necessary for this PR? Its getting pretty big as is. If so what is an example of what you are thinking?